app/components/ContextualDocsModal.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { offset, useFloating } from '@floating-ui/react'
+import { Popover } from '@headlessui/react'
+
+import { OpenLink12Icon, Question16Icon } from '@oxide/design-system/icons/react'
+
+import { ExternalLink } from './ExternalLink'
+
+type ContextualDocsLinkProps = {
+  href: string
+  linkText: string
+}
+
+export const ContextualDocsLink = ({ href, linkText }: ContextualDocsLinkProps) => (
+  <div className="border-b px-4 py-2 border-secondary last:border-0">
+    <ExternalLink
+      href={href}
+      className="rounded px-2 py-1 no-underline text-sans-md !text-default hover:bg-tertiary"
+    >
+      <span>{linkText}</span>
+      <OpenLink12Icon className="ml-1 translate-y-[1px] text-tertiary " />
+    </ExternalLink>
+  </div>
+)
+
+type ContextualDocsModalProps = {
+  heading: React.ReactNode
+  icon: JSX.Element
+  links: Array<ContextualDocsLinkProps>
+  summary: string
+}
+
+export const ContextualDocsModal = ({
```

Comment:
This isn’t a modal and contextual is kind of a mouthful — how about `DocsPopover`?

Reply:
Makes sense!